### PR TITLE
Switch to base 16 for TLS serial number in packetbeat in line with ECS changes

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## Proposed commit message
This switches the `serial_number` field in packetbeat's TLS reporting to use base 16 values instead of base 10:

```
		"public_key_size":             "2048",
		"serial_number":               "1121E97D5D37348C572C555A3A59B7B65D2B",
		"signature_algorithm":         "SHA256-RSA",
```

I don't think this is a breaking change? Also note the lack of a `0x` indicator. Not sure if we care.

ECS PR here: https://github.com/elastic/ecs/pull/2383

Also note that this maintains the liter bypass for `dsa`, which is deprecated. Do we still need it?

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

This will alter the reported `serial_number` for TLS certs from base 10 to base 16.

### CODE REVIEW

File: packetbeat/protos/tls/tls_test.go
Comment (nicholasberlin) [inline packetbeat/protos/tls/tls_test.go:427]:
```suggestion
```

File: packetbeat/protos/tls/parse.go
Comment (nicholasberlin) [inline packetbeat/protos/tls/parse.go:642]: I think it should be prefixed with `0x`, because I expect hex numbers to be that way. But, whatever.

**Review by nicholasberlin — APPROVED** [review]
LGTM if CI agrees

File: packetbeat/protos/tls/parse.go
Comment (fearful-symmetry) [reply packetbeat/protos/tls/parse.go:642]: same, but the ECS docs seem to imply it shouldn't have the `0x`?

```
Unique serial number issued by the certificate authority. For consistency, if this value is alphanumeric, it should be formatted without colons and uppercase characters.

type: keyword

example: 55FBB9C7DEBF09809D12CCAA
```

Comment (nicholasberlin) [reply packetbeat/protos/tls/parse.go:642]: Following the example makes sense to me.

**Review by andrewkroh — CHANGES_REQUESTED** [review]
Please update the PR description's proposed commit message to reference the related ECS change, as this provides essential context for understanding the behavior modification.

Has the ECS change been released? In my opinion, it is not advisable for Packetbeat to "get ahead" of the schema. First, we should create a release that incorporates the spec change. Additionally, the ECS version reported by Packetbeat ([src ref](https://github.com/elastic/beats/blob/930410642ebaf67a18c2a4e77256ac128b45bb21/packetbeat/cmd/root.go#L43)) should be updated to indicate which version it is following, even if it deviates from the other Beats.

We also need to add a changelog entry for this change.

**Comment by fearful-symmetry** [comment]
@andrewkroh so the ECS PR is here: https://github.com/elastic/ecs/pull/2383

But it doesn't look like there's been an actual release of this yet though. Should we wait for the next minor? Is there a release process for ECS we need to trigger? Looks like the last one was a year ago.

File: packetbeat/protos/tls/tls_test.go
Comment (andrewkroh) [inline packetbeat/protos/tls/tls_test.go:315]: Note that all of the fields under `tls.detailed.*` **are not** part of ECS. I cannot find any definition of this field within elastic/beats that needs updated, but it is defined at https://github.com/elastic/integrations/blob/7d7358618583e3d34e83e645d8bb7c83eda58253/packages/network_traffic/data_stream/tls/fields/protocol.yml#L210-L212 so that side will also need an update.

I recommend updating that definition at the same time in which you update the ECS version[^1][^2] for the network_traffic integration (after we have an ECS release).

[^1]: https://github.com/elastic/integrations/blob/7d7358618583e3d34e83e645d8bb7c83eda58253/packages/network_traffic/_dev/build/build.yml
[^2]: https://github.com/search?q=repo%3Aelastic%2Fintegrations+path%3A%2F%5Epackages%5C%2Fnetwork_traffic%5C%2Fdata_stream%5C%2F.*default%5C.yml%2F+%2Fecs%5C.version%2F&type=code

**Comment by andrewkroh** [comment]
> Should we wait for the next minor? Is there a release process for ECS we need to trigger?

According to https://github.com/elastic/ecs/blob/main/CONTRIBUTING.md#ecs-releases-during-the-donation-to-opentelemetry, it will need to be aligned to a stack minor. As for planning to get an actually release made, can you reach out to the `#ecs` slack channel to inquire about this (maybe we can do it for 8.16.0 so we don't have to wait long).

File: CHANGELOG.next.asciidoc
Comment (andrewkroh) [inline CHANGELOG.next.asciidoc:82]: I suggest adding some mention of ECS as the rational like "... to align with the ECS recommendation."

**Comment by andrewkroh** [comment]
Please update the example in your proposed commit message to reflect that the serial_number values are now uppercase.

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**
